src/gui/window_header.c

```diff
@@ -28,7 +28,7 @@ int16_t WINDOW_CLS_HEADER = 0;
 
 static void update_ETH_icon(bool link_up, window_header_t *window) {
     if (link_up) {
-        if (eeprom_get_var(EEVAR_LAN_FLAG).ui8 & LAN_EEFLG_TYPE) {
+        if ((eeprom_get_var(EEVAR_LAN_FLAG).ui8 & LAN_MSK_TYPE) == LAN_EEFLG_STATIC) {
```

Comment:
we need to create a file /common/wui_api.c/h where you define interface functions for WUI, there should be a function wui_eth_status(). 
here in GUI shall not be anything connected to wui

Reply:
yes, good point. I'm on it.